### PR TITLE
fixes #249: [4.x] BoltDriver does not support bolt+s and bolt+ssc

### LINKED ISSUE #249: [4.x] BoltDriver does not support `bolt+s` and `bolt+ssc`
This is from a Liquigraph example, using Hikari pool:
```java
Caused by: java.lang.RuntimeException: Driver org.neo4j.jdbc.bolt.BoltDriver claims to not accept jdbcUrl, jdbc:neo4j:bolt+s://HOST

	at com.zaxxer.hikari.util.DriverDataSource.<init>(DriverDataSource.java:110) ~[HikariCP-3.4.5.jar:na]
	at com.zaxxer.hikari.pool.PoolBase.initializeDataSource(PoolBase.java:325) ~[HikariCP-3.4.5.jar:na]
	at com.zaxxer.hikari.pool.PoolBase.<init>(PoolBase.java:114) ~[HikariCP-3.4.5.jar:na]
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:108) ~[HikariCP-3.4.5.jar:na]
```

Same happens with `bolt+ssc`.

I need to double check, but I think the `trust.strategy*` and `encryption` need to be **not** used when using any of the `neo4j` scheme.

### PR DESCRIPTION
**Pull request by conker84**
Implemented the support for `+s` and `+ssc` for both `neo4j` and `bolt` scheme

### CODE REVIEW

File: neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltDriver.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
  */
 public class BoltDriver extends BoltNeo4jDriverImpl {
 
-	public static final String JDBC_BOLT_PREFIX = "bolt";
+	public static final String JDBC_BOLT_PREFIX = "^bolt(\\+s|\\+ssc)*$";
```
Comment (fbiville) [inline neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltDriver.java]:
```suggestion
	public static final String JDBC_BOLT_PREFIX = "^bolt(\\+s|\\+ssc)?$";
```

File: neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/boltrouting/BoltRoutingNeo4jDriver.java [hunk 2]
```diff
@@ -40,8 +40,7 @@
  */
 public class BoltRoutingNeo4jDriver extends BoltNeo4jDriverImpl {
 
-//    public static final String JDBC_BOLT_ROUTING_PREFIX = "bolt\\+routing";
-    public static final String JDBC_BOLT_ROUTING_PREFIX = "neo4j";
+    public static final String JDBC_BOLT_ROUTING_PREFIX = "^neo4j(\\+s|\\+ssc)*$";
```
Comment (fbiville) [inline neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/boltrouting/BoltRoutingNeo4jDriver.java]:
```suggestion
    public static final String JDBC_BOLT_ROUTING_PREFIX = "^neo4j(\\+s|\\+ssc)?$";
```

**Review by fbiville — COMMENTED** [review]
Is the URL directly passed to the underlying driver?
If not, should there be code to automatically configure the trust strategy based on the scheme?

**Comment by conker84** [comment]
I updated the regex

> Is the URL directly passed to the underlying driver?

Yes the URL is passed to the driver, i.e. if we have `jdbc:neo4j:bolt://test` the driver will get `bolt://test`

**Pull request merged**